### PR TITLE
Enabled to list requests by approver's username

### PR DESCRIPTION
**Pull request by hsong-rh**
Enable to query requests based on approver, ex. `GET /requests?approver=jode@acme.com`

### CODE REVIEW

File: app/controllers/api/v1x0/requests_controller.rb [hunk 1]
```diff
@@ -14,11 +14,29 @@ def show
...
+        if params[:approver]
+          return index_by_approver
+        end
+
         Workflow.find(params.require(:workflow_id)) if params[:workflow_id] # to validate the workflow exists
         reqs = Request.includes(:stages).filter(params.slice(:requester, :decision, :state, :workflow_id))
         collection(reqs)
       end
 
+      def index_by_approver
```
Comment (bzwei) [inline app/controllers/api/v1x0/requests_controller.rb]: should move to a service class

File: app/controllers/api/v1x0/requests_controller.rb [hunk 2]
```diff
@@ -14,11 +14,29 @@ def show
       end
 
       def index
+        if params[:approver]
+          return index_by_approver
+        end
+
         Workflow.find(params.require(:workflow_id)) if params[:workflow_id] # to validate the workflow exists
```
Comment (bzwei) [inline app/controllers/api/v1x0/requests_controller.rb]: Should move to the same service class

File: app/controllers/api/v1x0/requests_controller.rb [hunk 3]
```diff
@@ -14,11 +14,29 @@ def show
       end
 
       def index
+        if params[:approver]
+          return index_by_approver
```
Comment (bzwei) [inline app/controllers/api/v1x0/requests_controller.rb]: why `return` here? shouldn't we support other filtering (decision, state etc) for an approver?
Please consider the filtering together with #104

Comment (hsong-rh) [reply app/controllers/api/v1x0/requests_controller.rb]: New changes are ready, will wait for #104 is merged.

**Comment by lindgrenj6** [comment]
Not sure if work is still being done on this - but you can use multiple filters on the URL e.g.

`GET /requests?filter[decisoin]=approved&filter[approver]=jdoe@acme.com`

would get all of the approved requests by a certain approver. Just looking at your tests it looks like that would be the main use for it.

More docs here: https://github.com/ManageIQ/manageiq-api-common#manageiqapicommonfilter

**Comment by hsong-rh** [comment]
@lindgrenj6 The filter has to be one of AR's attributes, right? Here approver is not `Request's` attribute. That's the reason to create this PR.

**Comment by lindgrenj6** [comment]
@hsong-rh yes that's correct, I didn't realize that approver wasn't an attribute on `Request`. My bad!

**Pull request merged**